components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1857,25 +1857,40 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
     async getStripeSetupIntentClientSecret(ctx: TraceContext): Promise<string | undefined> {
         const user = this.checkAndBlockUser("getStripeSetupIntentClientSecret");
         await this.ensureIsUsageBasedFeatureFlagEnabled(user);
-        const setupIntent = await this.stripeService.createSetupIntent();
-        return setupIntent.client_secret || undefined;
+        try {
+            const setupIntent = await this.stripeService.createSetupIntent();
+            return setupIntent.client_secret || undefined;
+        } catch (error) {
+            log.error("Failed to create Stripe SetupIntent", error);
+            throw new ResponseError(500, "Failed to create Stripe SetupIntent");
```

Comment:
Odd we don't have a 500 status code in our list of `errors.ts`

Reply:
Agree. We hard-code 500 in various places. Should probably fix that 👍 (in a follow-up 😅)